benches/centralized_telescope.rs

```diff
@@ -0,0 +1,201 @@
+//! Benchmarking centralized_telescope
+
+use criterion::{black_box, criterion_group, criterion_main, measurement::WallTime, Criterion};
+use rand_chacha::ChaCha20Rng;
+use rand_core::{RngCore, SeedableRng};
+use std::time::{Duration, Instant};
+
+use alba::centralized_telescope::{algorithm, init, params::Params, setup::Setup};
+
+pub mod criterion_helpers;
+use criterion_helpers::{benchmarks, Repetitions, Steps};
+
+#[path = "../src/utils/test_utils.rs"]
+mod test_utils;
+
+// Global variables
+const NAME: &str = "Centralized";
+// Benchmark parameters
+const L: &[f64] = &[50.0, 128.0]; // Security parameter
+const SP: &[u64] = &[1_000]; // Size of set to lower bound
+const NP: &[u64] = &[80, 90, 95, 98]; // Alba's np parameter, |Sp| >= np
+const NF: &[u64] = &[67, 75]; // Alba's nf parameter,  |Sp| >= np > nf
+
+/// Function generating a random set of elements to bench and calling Alba's centralized setup
+pub fn centralized_setup(
+    rng: &mut ChaCha20Rng,
+    l: f64,
+    sp: u64,
+    np: u64,
+    nf: u64,
+) -> (Vec<[u8; 32]>, Setup) {
+    let seed_u32 = rng.next_u32();
+    let seed = seed_u32.to_ne_bytes().to_vec();
+    let dataset: Vec<[u8; 32]> = test_utils::gen_items(&seed, sp);
+    let params = Params {
+        soundness_param: l,
+        completeness_param: l,
+        set_size: np.saturating_mul(sp).div_ceil(100),
+        lower_bound: nf.saturating_mul(sp).div_ceil(100),
+    };
+    (dataset, init::make_setup(&params))
+}
+
+/// Bench the duration of both the proving and verifiying algorithm of Alba centralized
+fn time_benches(c: &mut Criterion) {
+    #[allow(clippy::unit_arg)]
+    fn prove_duration(l: f64, sp: u64, np: u64, nf: u64, truncate_size: u64, n: u64) -> Duration {
+        let mut rng = ChaCha20Rng::from_entropy();
+        let mut total_duration = Duration::ZERO;
+        for _ in 0..n {
+            // Setup
+            let (mut dataset, bench_setup) = centralized_setup(&mut rng, l, sp, np, nf);
```

Comment:
I was an asked from Arnaud from a long time ago.

Basically you do not want to overfit your bench on a particular dataset so you generate several of them instead.

Reply:
But then the graphs don't show what I would expect. For example, I was looking at the graphs for the number of repetitions bench (when it works), and noticed that the distribution of the measured value shows two values: 1 and 1.25. What happens is, every sample has 4 iterations and some samples obtain values (1, 1, 1, 2) with the average of 5/4. Is this behavior intended? I was expecting to see the real distribution.

Reply:
I have never seen repetition graph as it failed every single time so I am not sure why you saw this. 
This behaviour is not explained on criterion [docs](https://bheisler.github.io/criterion.rs/book/user_guide/timing_loops.html). The benchmarks for step and time seemed ok to me

Reply:
But you get what I'm saying. Each sample is actually an average of multiple measurements, so the graph would be concentrated around the average much more than the distribution actually is. Applies to step and time as well.

Reply:
That shouldn't be a problem though as it will remove some variance due to background processes or other.

What matters is that we have the trend for any data for the same given parameters which is what is benched here.

Reply:
If we only care about measuring the average, it's fine.